### ISSUE TITLE
Display more data in the app (stand-alone without upload)?

### ISSUE THREAD

**Issue by psit** [pazaan/600SeriesAndroidUploader#300]
I'm looking for a stand-alone app without data upload to view the 640G pump data.
(I already got some info and had a great start in this issue: https://github.com/pazaan/600SeriesAndroidUploader/issues/296)

Now I am wondering if there is a way to show some more data from the pump in either the uploader app or xdrip+ (sadly the xdrip+ documenation is very sparse, but the data is provided by the uploader, so I start asking here), namely:

- current alert message from the pump, if not dismissed
- especially "basal interruption" (possibly with indicator in the timeline about the interruption)
- carbohydrates, bolus (also delayed and dual) and "event markers" in the timeline/graph
- basal rate in the timeline
- blood glycose measurements (sent to the pump via Contour Next Link, partially used for calibration) in the timeline/graph

Eventually, it would be great to have all data from the medtronic carelink daily reports graphs, available on a smartphone.
I had a short look at the https://github.com/pazaan/decoding-contour-next-link project, which seems related to the android uploader (thanks @pazaan). There, the mentioned data seems to be available.

Thanks a lot for your effort to everyone! You are improving our lives drastically!

**Comment by Pogman** [comment]
The nightscout web server combined with the uploader provides all of this. I would highly recommend getting that set up and tweaked to your needs.

**Comment by psit** [comment]
Is this the place where all data is sent to xdrip+?
https://github.com/pazaan/600SeriesAndroidUploader/blob/6cc7acd9e69ad9c4f03130afffebce095e3e73e8/app/src/main/java/info/nightscout/android/xdrip_plus/XDripPlusUploadService.java#L175
I'm wondering if one could send more information to xdrip?

If I ever have some time left I could also try contributing as a developer.

**Comment by psit** [comment]
Is there some API or any information about how to send more data to xDrip+? What format do they expect?

**Comment by pazaan** [comment]
The nice thing is that xDrip+ is also Open Source, so you can look through their code to figure it out 😄

**Comment by psit** [comment]
Yes, I know - and with so many great projects that are under heavy development, nobody cares about documentation... I have this in my daily business ;) I was hoping that someone has a clue about this interface between the apps.
Sadly I am just a Java dev with no clue about Android apps. I guess I will have to do some learning if I ever find time for it.

**Comment by psit** [comment]
Does anybody know if this is the recieving end of the xDrip data broadcast?
https://github.com/NightscoutFoundation/xDrip/blob/19bc5be65f0bc255cd07b8f8d35cc387a823bc3a/app/src/main/java/com/eveningoutpost/dexdrip/NSEmulatorReceiver.java
